asmcomp/selectgen.ml

```diff
@@ -851,9 +851,9 @@ method emit_expr (env:environment) exp =
               let loc_res =
                 self#insert_op_debug env new_op dbg
                   loc_arg (Proc.loc_external_results rd) in
-              self#insert_move_results env loc_res rd stack_ofs;
+              if returns then self#insert_move_results env loc_res rd stack_ofs;
```

Comment:
Is this needed ? I would have thought giving mach type `typ_void` would mean that the problem doesn't apear

Reply:
it's not needed, and indeed if the call has a return type of ` typ_void`  then no instructions will be emitted.
I had added this as a over-zealous optimization for cases where a non returning call with non void return type would be used, but I agree that it would probably be an error for such a call to exist (un_cps takes care of enforcing that invariant but there is no easy way for cmmgen).
In any case, i removed that optimization, as it's not really needed, and the code is simpler without it.